### DISCUSSION TITLE
Can this be used with a mono-repo or multiple module project?

### DISCUSSION THREAD

**Discussion by sheldonhull** [restechnica/semverbot#19] [Q&A] [answered]
Since the project is based on Git commits and history, I'm assuming it isn't designed for use with mono repo or multiple-modules in a single repository.

Can you confirm this tool would focus on a single repository being a single deployable & versionable release?

**Comment by shiouen** [comment, marked as answer]
I don't often work with mono repos, but it depends on your workflow. There's so many options I'm not entirely sure where to begin.

Most important with `sbot`: whatever you do before raising a PR doesn't really matter, as long as you move along all commits that represent a change within a single PR. `sbot` only looks at the latest git commit for information.

The mono repo could work if you keep changes to a single branch and share that branch with multiple devs. It could also work when you consolidate all commits part of a change (release, feat, fix) into a temporary branch and then merge the whole thing with a pull request into main. `sbot` would simply work with the PR message or branch name of the temporary branch.

Haven't used semverbot with Git modules yet, I will have to try it first.
